### PR TITLE
Fixed #8603 - Resolved SQL exception and fixed translation-related is…

### LINKED ISSUE #8603: SQL error occurs when clicking on Print in the frontend invoice
### Terms

- [X] Before you openning this issue, i have checked if the issue has already been reported.

### Bagisto Version(s) affected

master

### Issue Description

SQL error occurs when clicking on Print in the frontend invoice

### Preconditions

-  A user should be logged in on the user panel.
- The user needs to place an order, and an invoice should be generated by the admin for that order.

### Steps to reproduce

1. Navigate to the 'User panel.'
2. Place an order.
3. Access the 'Admin panel.'
4. Click on 'Sales' > 'Orders.'
5. Open the specific order and create an invoice.
6. Return to the 'User panel.'
7. Go to 'Home' > 'My Account' > 'Orders.'
8. Open the order and click on the invoice.
9. Then, click on 'Print.'
10. See the error.

### Expected Result

At this point, the invoice should ideally be printed or downloaded as a PDF.

### Actual Result

An error message is prompting to run php artisan migrate, but running it does not resolve the problem.

### Screenshots

Screenshot from 2023-10-04 17-49-57

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sunarcrajneesh**
…sues in the invoice PDF.

## Issue Reference
Fixed #8603 - SQL error occurs when clicking on Print in the frontend invoice.

## Description

- An additional field was being used in the OrderController to fetch invoices, but it was not created in the table, removed that.
- After making changes in the OrderController, invoices could be downloaded. however, there were translation issues present, which have also been resolved now.

## How To Test This?

- After pulling the changes, navigate to 'Home' > 'My Account' > 'Orders' Then, open an order for which the invoice has been generated. Click on the 'invoive' > 'print' button. it may take some time,and it will download the PDF of the invoice. 
- Can also test this with a new order.

## Documentation
- [ ] My pull request requires an update on the documentation repository.

### CODE REVIEW

File: packages/Webkul/Shop/src/Resources/views/customers/account/orders/pdf.blade.php [hunk 1]
```diff
@@ -192,7 +192,7 @@
 
                                 <div class="row">
                                     <span
-                                        class="label">@lang('shop::app.customer.account.order.view.invoice-date'):
+                                        class="label">@lang('admin::app.sales.invoices.invoice-pdf.date'):
```
Comment (amit-webkul) [inline packages/Webkul/Shop/src/Resources/views/customers/account/orders/pdf.blade.php:195]: Same here

Comment (sunarcrajneesh) [reply packages/Webkul/Shop/src/Resources/views/customers/account/orders/pdf.blade.php:195]: @amit-webkul,
We have used the 'admin' lang in most places. Do we need to update it everywhere, or is it necessary only for those two specific locations you mentioned?

Comment (devansh-webkul) [reply packages/Webkul/Shop/src/Resources/views/customers/account/orders/pdf.blade.php:195]: For separation of concern, in the shop package, there will be no admin-related lang should be present. Same vise versa for admin also.

Comment (devansh-webkul) [reply packages/Webkul/Shop/src/Resources/views/customers/account/orders/pdf.blade.php:195]: If specific language keys are not present then add your key to the shop package also.

**Comment by sunarcrajneesh** [comment]
Added required language keys in the shop package and utilized them in the blade files.

File: packages/Webkul/Shop/src/Resources/views/customers/account/orders/pdf.blade.php [hunk 2]
```diff
@@ -171,7 +171,7 @@
                             </div>
                         @endif
                         <div class="invoice-text">
-                            <span>{{ strtoupper(__('admin::app.sales.invoices.invoice')) }}</span>
+                            <span>{{ strtoupper(__('shop::app.customers.account.orders.view.invoices.invoice-pdf.invoice')) }}</span>
```
Comment (amit-webkul) [inline packages/Webkul/Shop/src/Resources/views/customers/account/orders/pdf.blade.php:174]: Please re-arrange the '**lang key**' for all lines to match the folder structure, like this:
Like this :-  **('shop::app.customers.account.orders.invoice-pdf')**

File: packages/Webkul/Shop/src/Resources/lang/en/app.php [hunk 3]
```diff
@@ -202,6 +202,32 @@
                         'shipping-handling'  => 'Shipping & Handling',
                         'tax-amount'         => 'Tax Amount',
                         'tax'                => 'Tax',
+                        'invoice-pdf' => [
```
Comment (amit-webkul) [inline packages/Webkul/Shop/src/Resources/lang/en/app.php:205]: Please add a new line space here, and arrange the keys in alphabetical order based on the first letter of each key.

File: packages/Webkul/Shop/src/Resources/lang/uk/app.php [hunk 4]
```diff
@@ -201,7 +201,6 @@
                         'subtotal'           => 'Промежуточный итог',
                         'shipping-handling'  => 'Доставка и обработка',
                         'tax-amount'         => 'Сумма налога',
-                        'tax'                => 'Налог',
```
Comment (amit-webkul) [inline packages/Webkul/Shop/src/Resources/lang/uk/app.php:204]: The tax key has been removed. Please add it back.

File: packages/Webkul/Shop/src/Resources/lang/ru/app.php [hunk 5]
```diff
@@ -201,7 +201,6 @@
                         'subtotal'           => 'Промежуточная сумма',
                         'shipping-handling'  => 'Доставка и обработка',
                         'tax-amount'         => 'Сумма налога',
-                        'tax'                => 'Налог',
```
Comment (amit-webkul) [inline packages/Webkul/Shop/src/Resources/lang/ru/app.php:204]: The tax key has been removed. Please add it back

**Comment by sunarcrajneesh** [comment]
Updated translation for 'invoices' array in hindi and added missing key.

File: packages/Webkul/Shop/src/Resources/lang/uk/app.php [hunk 6]
```diff
@@ -201,7 +201,7 @@
...
+                        'tax'               => 'Налог',
```
Comment (amit-webkul) [inline packages/Webkul/Shop/src/Resources/lang/uk/app.php:204]: Kindly add one space before the (=>) symbol.

File: packages/Webkul/Shop/src/Resources/lang/ru/app.php [hunk 7]
```diff
@@ -201,7 +201,7 @@
...
+                        'tax'               => 'Налог',
```
Comment (amit-webkul) [inline packages/Webkul/Shop/src/Resources/lang/ru/app.php:204]: Kindly add one space before the (=>) symbol.

Comment (sunarcrajneesh) [reply packages/Webkul/Shop/src/Resources/lang/ru/app.php]: One space Added.

File: packages/Webkul/Shop/src/Resources/lang/uk/app.php (hunk 6 above)
Comment (sunarcrajneesh) [reply packages/Webkul/Shop/src/Resources/lang/uk/app.php]: One space Added.

**Review by amit-webkul — APPROVED** [review]

**Comment by Amitk-webkul** [comment]
@sunarcrajneesh Not getting the order ID and Invoice ID in the French language. [Her](https://prnt.sc/fK3xVt0ueFv4)

In the Russian language order ID translation is not correct. [Here](https://prnt.sc/DDFa2Pz3rkb9)

**Comment by sunarcrajneesh** [comment]
@Amitk-webkul,
On  "Home > My Account > Orders > View"  the term "Order #" is being used with "Order_id" as a title. So according to the Russian language, the translation for "Order" is correct as "Заказ."

**Comment by Amitk-webkul** [comment]
@sunarcrajneesh That's correct! The translation for 'Order' in Russian is indeed 'Заказ'. Please proceed to another issue.

**Comment by Amitk-webkul** [comment]
@sunarcrajneesh Now it is working fine.

**Review by Amitk-webkul — APPROVED** [review]

**Review by amit-webkul — APPROVED** [review]

File: packages/Webkul/Shop/src/Http/Controllers/Customer/Account/OrderController.php [hunk 8]
```diff
@@ -63,7 +63,6 @@ public function printInvoice($id)
     {
         $invoice = $this->invoiceRepository->findOneWhere([
             'id'          => $id,
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/Customer/Account/OrderController.php:65]: Why customer ID removed from here?

**Review by devansh-webkul — CHANGES_REQUESTED** [review]
Please add a join query with the order and add back the `customer_id` check.

File: packages/Webkul/Shop/src/Http/Controllers/Customer/Account/OrderController.php (hunk 8 above)
Comment (sunarcrajneesh) [reply packages/Webkul/Shop/src/Http/Controllers/Customer/Account/OrderController.php]: Refactored invoice retrieval to include order customer inquiry.

File: packages/Webkul/Shop/src/Http/Controllers/Customer/Account/OrderController.php [hunk 9]
```diff
@@ -61,10 +61,11 @@ public function view($id)
      */
     public function printInvoice($id)
     {
-        $invoice = $this->invoiceRepository->findOneWhere([
-            'id'          => $id,
-            'customer_id' => auth()->guard('customer')->id(),
-        ]);
+        $invoice = $this->invoiceRepository->where('id', $id)
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/Customer/Account/OrderController.php:64]: You can add `firstOrFail` and reduce the next three lines check.

Comment (sunarcrajneesh) [reply packages/Webkul/Shop/src/Http/Controllers/Customer/Account/OrderController.php:64]: @devansh-webkul,
Are you saying that the customer doesn't need to check here?

Comment (devansh-webkul) [reply packages/Webkul/Shop/src/Http/Controllers/Customer/Account/OrderController.php:64]:
~~~php
$invoice = $this->invoiceRepository->where('id', $id)
            ->whereHas('order', function ($query) {
                $query->where('customer_id', auth()->guard('customer')->id());
            })
            ->firstOrFail();
~~~

**Review by devansh-webkul — APPROVED** [review]

**Review by jitendra-webkul — APPROVED** [review]

**Pull request merged**